subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestSuiteTestFrameworkIntegrationTest.groovy

```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.DefaultTestExecutionResult
+import org.gradle.test.fixtures.file.TestFile
+
+class TestSuiteTestFrameworkIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
The Test Suite related test classes are getting pretty scattered in this project, some in the `testsuites` package, some not. 

It looks like this class might be better being split into one class per Test Framework, with a common base class, or use `expect` blocks to consolidate some of these similar tests.

Also there is a Spock test `BuildSrcSpockIntegrationTest` in the `junit` pacakge.

Just seems like the tests could use some further organizing for this project.

Reply:
What's the expected behavior when a test task is configured directly, outside of a Suite, after being configured in a Suite?

Reply:
> What's the expected behavior when a test task is configured directly, outside of a Suite, after being configured in a Suite?

Agreed - what happens in this scenario?

```
testing {
  suites {
    test {
      useJUnit()
      targets.all {
        testTask.configure {
          options {
            // configure something
          }
        }
      }
    }
  }
}

test {
  useTestNG()
}
```

Reply:
good suggestions, I can take a stab at this tomorrow

Reply:
I've moved around a few tests that were new for test suites. I also looked around at the existing tests and I agree they're messy. I don't want to hold up 7.3 to fix them, but it looks like there's a bit of simplification we could do here.

I've added the test that Kyle suggested